### ISSUE TITLE
Add throttleLatest Flow operator

### ISSUE THREAD

**Issue by fluidsonic** [Kotlin/kotlinx.coroutines#1927]
We have no operator that prevents subsequent emissions for a certain time after an emission.
In RxJava there is `throttleLatest` for that.

```kotlin
flow {
    repeat(10) {
        emit(it)
        delay(101)
    }
}
    .throttleLatest(500)
    .collect {
        println(it)
    }
```
The example above should print `0` and `5`. Both should be emitted without any added delay.

One **use-case** is button clicks where (accidental) repeated clicks should be filtered. If I click a button twice accidentally, only the first click event would make it downstream and the second click be ignored because it came in too fast after the first one.

`debounce(500)` is not suitable because it
a) delays the emission of the first value by least 500ms and
b) delays the emission of any value until the user stops clicking for at least 500ms.

`sample(500)` is not suitable because it
a) delays the emission of the first value by least 500ms and
b) emits nothing if the `Flow` is closed within 500ms after collection has started.

`throttleLatest` was also mentioned in #1107 but no reason was given why it didn't make it into `Flow` so far.

**Comment by jxdabc** [comment]
@fluidsonic 
If I'm not wrong.
The above `throttleLatest` example should print `0`, '4'.
The operator you want seems being called `throttleFirst`.
Plese see https://github.com/ReactiveX/RxJava/wiki/Filtering-Observables#throttlefirst

**Comment by fluidsonic** [comment]
@jxdabc you're right, `throttleFirst` is more appropriate for solving the problem. I'll update the issue.

**Comment by davidmigloz** [comment]
Duplicate of https://github.com/Kotlin/kotlinx.coroutines/issues/1927

**Comment by hoc081098** [comment]
please check my lib https://github.com/hoc081098/FlowExt https://hoc081098.github.io/FlowExt/docs/latest/-flow-ext/com.hoc081098.flowext/throttle-time.html
